### DISCUSSION TITLE
A way to list all packages in the monorepo

### DISCUSSION THREAD

**Discussion by vjpr** [pnpm/pnpm#8690] [Ideas]
There are many tools that rely on getting the list of all workspace dependencies and other dependency related tasks.

There seem to be many pnpm packages to help with this but they are undocumented, duplicated, deprecated, or require a bunch of config/options.

E.g.

- @pnpm/list
- @pnpm/reviewing.dependencies-hierarchy
- @pnpm/workspace.filter-packages-from-dir
- @pnpm/find-workspace-packages
- @pnpm/filter-workspace-packages

We need a clean way to read packages, filter packages, list workspace packages, list dependencies.

It should have good DX.

Maybe there should be a separation of private api and public api packages?

Maybe something like `@pnpm/api.<blah>` to indicate a programmatic public api.

# Related

https://github.com/pnpm/pnpm/issues/3482

https://github.com/teambit/bit/blob/master/scopes/dependencies/pnpm/lynx.ts

**Comment by vjpr** [comment]
This looks like how Bit does it...

https://github.com/teambit/bit/blob/10ac015e9cb96a7f9438571ea993bb579ebf09ae/scopes/dependencies/pnpm/pnpm.package-manager.ts#L305

**Comment by canadaduane** [comment]
See also https://github.com/pnpm/pnpm/issues/1519#issuecomment-1299922699
